src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/scaleset/AzureScaleSetProvider.java

```diff
@@ -0,0 +1,327 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.connector.iaas.cloud.provider.azure.scaleset;
+
+import java.util.*;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.log4j.Logger;
+import org.ow2.proactive.connector.iaas.cloud.provider.azure.AzureProvider;
+import org.ow2.proactive.connector.iaas.model.*;
+import org.springframework.stereotype.Component;
+
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.compute.*;
+import com.microsoft.azure.management.network.*;
+import com.microsoft.azure.management.network.Network;
+import com.microsoft.azure.management.resources.ResourceGroup;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+
+import lombok.Getter;
+
+
+/**
+ * Provides Microsoft Azure clouds' management using the official java SDK.
+ *
+ * This class has been tested by ActiveEon to be thread safe (using Azure SDK release version 1.0.0-beta5).
+ * However this need to be carefully double-checked after every SDK upgrades, as mentioned by Microsoft:
+ * ------------------------------------------------------------------------------------------------------------
+ * We do not make any thread-safety guarantees about our libraries. We also do not test them for thread-safety.
+ * Methods that are currently thread-safe may be thread-unsafe in future versions.
+ * ------------------------------------------------------------------------------------------------------------
+ *
+ * @author ActiveEon Team
+ * @since 01/03/17
+ */
+@Component
+public class AzureScaleSetProvider extends AzureProvider {
+
+    /*
+     * Assumptions for the following code:
+     *
+     * 1 NodeSource (infra) <-> 1 scale set </-> 1 resource group
+     *
+     */
+
+    @Getter
+    protected final String type = "azureScaleSet";
+
+    private final Logger logger = Logger.getLogger(AzureScaleSetProvider.class);
+
+    @Override
+    public Set<Instance> createInstance(Infrastructure infrastructure, Instance instance) {
+
+        // Initialize Azure service connector
+        Azure azureService = azureServiceCache.getService(infrastructure);
+
+        // Retrieve general information, tags & options
+        String instanceTag = Optional.ofNullable(instance.getTag())
+                                     .orElseThrow(() -> new RuntimeException("ERROR missing instance tag/name from instance: '" +
+                                                                             instance + "'"));
+        List<Tag> tags = tagManager.retrieveAllTags(instance.getOptions());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "tags" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1481)